### PR TITLE
GH-1484 sparqlstar

### PR DESCRIPTION
**Pull request by damyan-ognyanov**
Adds extensions to SparqlParser supporting SPARQL* 
* introduces two additional sparql algebra nodes `TripleRef` and `ValueExprTripleRef`
* allows use of SPARQL* embedded triple expressions in `BIND`, `VALUES`, and subject or object of a `StatementPattern`
* allows to use these also in `CONSTRUCT`, `INSERT` and `DELETE`
* added 5 new functions to work with `Triple` values - these reuse the rdf reification vocabulary
* added method to evaluate `ValueExprTripleRef` in `StrictEvaluationStrategy`. 
* currently there is no implementation of `evaluate(TripleRef)` because it is still unclear whether the default evaluation should lookup over the nodes introduced through standard reification or we need an extention to `TripleSource` that allow for lookup over the nodes of type `Triple` in the undelying Model/Sail

There is a basic test suite covering the extensions in `TestSparqlStarParser`. Each case there has the expected query model as comment above the testcase.

### CODE REVIEW

File: core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunction.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunction.java]: This is not the correct copyright header I'm afraid. It should be this:
```

/*******************************************************************************
 * Copyright (c) 2020 Eclipse RDF4J contributors.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.php.
 *******************************************************************************/
```

See https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting for details.

File: core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/SyntaxTreeBuilderConstants.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018 Eclipse RDF4J contributors, Aduna, and others.
+ * Copyright (c) 2020 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (jeenbroekstra) [inline core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/SyntaxTreeBuilderConstants.java]: No big dealm but there's no need to update copyright years in existing files. We never update these: they reflect the year the file was first created, not when it was last edited.

File: core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunction.java (hunk 1 above)
Comment (damyan-ognyanov) [reply core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunction.java]: Updated the incorrect copyright headers

File: core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/SyntaxTreeBuilderConstants.java (hunk 2 above)
Comment (damyan-ognyanov) [reply core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/SyntaxTreeBuilderConstants.java]: Updated the headers fro the existing files e.g. changed back to `2018` from `2020` ...

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunctionTest.java]: You missed one I'm afraid.

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/StatementFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,239 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/StatementFunctionTest.java]: This one too.

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/TripleObjectFunctionTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors, Aduna, and others.
```
Comment (jeenbroekstra) [inline core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/TripleObjectFunctionTest.java]: And this one.

File: core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunctionTest.java (hunk 3 above)
Comment (damyan-ognyanov) [reply core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/function/triple/IsTripleFunctionTest.java]: updated the rest of the headers and squashed all commits into one

**Pull request merged**